idaes/surrogate/alamopy_new.py

```diff
@@ -0,0 +1,504 @@
+    "ignore",
+    "xisint",
+    "zisint",
+    "tolrelmetric",
+    "tolabsmetric",
+    "tolmeanerror",
+    "tolsse",
+    "mipoptca",
+    "mipoptcr",
+    "linearerror",
+    "GAMS",
+    "GAMSSOLVER",
+    "solvemip"]
+
+"""
+Excluded Options:
+NSAMPLE, NVALSAMPLE, INITIALIZER, PRINT TO FILE, FUNFORM, NTRANS
+PRINT TO SCREEN - try to handle through logger
+
+Simulator options (not yet supported):
+MAXSIM, MINPOINTS, MAXPOINTS, SAMPLER, SIMULATOR, PRESET, TOLMAXERROR, SIMIN,
+SIMOUT
+
+Other options not yet included:
+NCUSTOMBAS
+"""
+
+
+class Alamopy(Surrogate):
+    """
+    Standard SurrogateModelTrainer for ALAMO.
+
+    This mainly defines a set of configuration options for ALAMO along with
+    methods to read and write the ALAMO input and poutput files and to call
+    the ALAMO executable.
+
+    Generally, options default to None to indicate that no entry will be
+    written in the ALAMO input file. In this case, the default ALAMO settings
+    will be used.
+    """
+    # The follwoing ALAMO options are not (yet) supported
+    # Returning model prediction, as we can do that in IDAES
+    # Ininital sampling of data, as SurrogateModelBuilder should do that
+    # Single valdiation set, due to limitations of current API
+    # Adaptive sampling is not yet implemented
+    # Custom basis functions are not yet implemented
+
+    CONFIG = Surrogate.CONFIG()
+
+    CONFIG.declare('xfactor', ConfigValue(
+        default=None,
+        domain=list_of_floats,
+        description="List of scaling factors for input variables."))
+    CONFIG.declare('xscaling', ConfigValue(
+        default=None,
+        domain=In([True, False]),
+        description="Option to scale input variables.",
+        doc="Option to scale input variables. If True and xfactors are not "
+        "provided, ALAMO sets XFACTORS equal to the range of each input "
+        "variable."))
+    CONFIG.declare('scalez', ConfigValue(
+        default=None,
+        domain=In([True, False]),
+        description="Option to scale output variables."))
+
+    # Basis function options
+    CONFIG.declare('monomialpower', ConfigValue(
+        default=None,
+        domain=list_of_ints,
+        description="Vector of monomial powers considered in basis "
+        "functions."))
+    CONFIG.declare('multi2power', ConfigValue(
+        default=None,
+        domain=list_of_ints,
+        description="Vector of powers to be considered for pairwise "
+        "combinations in basis functions."))
+    CONFIG.declare('multi3power', ConfigValue(
+        default=None,
+        domain=list_of_ints,
+        description="Vector of three variable combinations of powers to be "
+        "considered as basis functions."))
+    CONFIG.declare('ratiopower', ConfigValue(
+        default=None,
+        domain=list_of_ints,
+        description="Vector of ratio combinations of powers to be considered "
+        "in the basis functions."))
+    CONFIG.declare('constant', ConfigValue(

```

Comment:
I like that linear and constant terms are assumed, similar to the GUI. This should be noted in the documentation, though, since it's not immediately obvious to new users.